app/models/rent.rb

```diff
@@ -3,4 +3,34 @@ class Rent < ApplicationRecord
   belongs_to :book, optional: true
 
   validates :start_date, :end_date, presence: true
+  validate :check_creation
+
+  scope :from_today, -> { where('DATE(?) BETWEEN start_date AND end_date', Time.zone.today) }
+
+  def reserve_conflict?
+    if book_id
+      rents = Rent.from_today.where(book_id: book_id)
```

Comment:
i'm looking something like
```
Rent.where(book_id: book.id).where('DATE(:start_date) BETWEEN start_date AND end_date OR'\
                                   'DATE(:end_date) BETWEEN start_date AND end_date OR'\
                                   'start_date BETWEEN :start_date AND :end_date OR'\
                                   'end_date BETWEEN :start_date AND :end_date',
                                   start_date: start_date, end_date: end_date)
```

Reply:
I'm tying to use 'scope :from_today, -> { where('DATE(?) BETWEEN start_date AND end_date', Time.zone.today) }' to achieve records with the same scope in 'rents' and 'books'